homeassistant/components/geniushub/config_flow.py

```diff
@@ -0,0 +1,173 @@
+"""Config flow for Geniushub logger integration."""
+
+from __future__ import annotations
+
+from http import HTTPStatus
+import logging
+import socket
+from typing import Any
+
+import aiohttp
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlowResult
+
+from . import V1_API_SCHEMA, V3_API_SCHEMA, validate_input
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Solis logger."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """User config step for determine v1 or v3."""
+        _LOGGER.debug("ConfigFlow.async_step_user")
+        return self.async_show_menu(
+            step_id="user",
+            menu_options=["option_1", "option_2"],
+        )
```

Comment:
Can you maybe explain what V1 and V3 is? I am assuming that it's an API version, but how does the end user know this? Are there certain types of devices that work with one but not the other?

In any case, can we pick the V1 and V3 names (or another recognizable name that's for that version) instead of option 1 or 2? This is one of those things you have to know before you can understand what this code does

Reply:
V1 is the cloud-hosted API that the company provide.

V3  is the direct connection to your local hardware. It also returns more data.

One also requires your Genius Hub username and password, whilst the other works with a token you can generate on their website.

So, V3 is the preferred API to use, as long as your Home Assistant device can talk directly to your Genius hub.

Reply:
Would it make sense to call it Cloud and Local?

Reply:
Possibly, though I suppose there's nothing to stop someone setting up their home network so that they can remotely access their Genius hub over the internet.

Maybe Cloud and Direct would be better?

Reply:
I mean, that is possible with a lot of are called local within home assistant

Reply:
"Option 1" and "Option 2" come from the existing documentation.

This is what the user sees:
```
 "menu_options": {
          "option_1": "Local Hub hostname/address with user credentials",
          "option_2": "Cloud Hub token only"
        }
```

Reply:
Yes, but we also want to make the code readable, and I'm not a geniushub user, so option 1 and option 2 don't really tell me something